src/charm.py

```diff
@@ -1,88 +1,161 @@
 #!/usr/bin/env python3
-# Copyright 2023 nikos
+# Copyright 2023 bence
 # See LICENSE file for licensing details.
 #
 # Learn more at: https://juju.is/docs/sdk
 
 import logging
-
+from charms.observability_libs.v0.kubernetes_service_patch import KubernetesServicePatch
 from charms.traefik_k8s.v1.ingress import (
+    IngressPerAppReadyEvent,
     IngressPerAppRequirer,
+    IngressPerAppRevokedEvent,
+)
+from ops.charm import (
+    CharmBase,
+    ConfigChangedEvent,
+    HookEvent,
+    WorkloadEvent,
 )
-from charms.observability_libs.v0.kubernetes_service_patch import KubernetesServicePatch
-from ops.charm import CharmBase
 from ops.main import main
-from ops.model import ActiveStatus, MaintenanceStatus
+from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus, ModelError, WaitingStatus
+from ops.pebble import ChangeError, Layer
 
-# Log messages can be retrieved using juju debug-log
 logger = logging.getLogger(__name__)
 
-VALID_LOG_LEVELS = ["info", "debug", "warning", "error", "critical"]
-
 
-class KratosUiOperatorCharm(CharmBase):
+class IdentityPlatformLoginUiOperatorCharm(CharmBase):
     """Charm the service."""
 
     def __init__(self, *args):
         super().__init__(*args)
-        self._container_name = "kratos-ui"
+        self._container_name = "login_ui"
+        self._container = self.unit.get_container(self._container_name)
+        """Init attributes"""
+        self._hydra_url = None
+        self._kratos_url = None
+        self._port = None
+        self._reconfigure()
```

Comment:
we don't need to put those in init, you can straight away get them from config where needed